### PR TITLE
Update cherry-pick docs for cloud provider-specific changes

### PR DESCRIPTION
**Pull request by feiskyer**
**Which issue(s) this PR fixes**:

Fixes #

Since out-of-tree cloud providers are not GA yet, to ensure all active Kubernetes versions running well on in-tree cloud providers, we should allow cherry-picks for cloud provider-specific changes (e.g. cloud API version upgrading or small features limited to one single cloud provider).

cc @tpepper @justaugustus

### CODE REVIEW

**Comment by tpepper** [comment]
-1

I don't think the community can sustain the risk and effort of accepting changes from cloud providers with zero constraints.

I think the text removed in the PR, ie:

```
If the change is in cloud provider-specific platform code (which is in the process of being moved out of core Kubernetes), describe the customer impact, how the issue escaped initial testing, remediation taken to prevent similar future escapes, and why the change cannot be carried in your downstream fork of the Kubernetes project branches.
```

should remain and be supplemented by additional information on:

 * how cloud-provider change proposals will be evaluated relative to risk

**Comment by feiskyer** [comment]
@tpepper thanks for the comments. Updated per the suggestions above. Could you look at them again?

File: contributors/devel/sig-release/cherry-picks.md [hunk 1]
```diff
@@ -61,7 +61,13 @@ If the change is in cloud provider-specific platform code (which is in the
 process of being moved out of core Kubernetes), describe the customer impact,
 how the issue escaped initial testing, remediation taken to prevent similar
 future escapes, and why the change cannot be carried in your downstream fork of
-the Kubernetes project branches.
+the Kubernetes project branches. The changes are only allowable when:
+
+- No new features are introduced (e.g. changing Kubernetes API or supporting new
+  Kubernetes features)
```
Comment (feiskyer) [inline contributors/devel/sig-release/cherry-picks.md:67]: @tpepper do you think "adding a new annotation" is adding a new feature? annotations are pretty useful to enable or disable things while won't break other scenarios.

Comment (tpepper) [reply contributors/devel/sig-release/cherry-picks.md:67]: Adding a default annotation into k/k?  I'd like to think of cloud provider specific annotations as an operational pattern and something that wouldn't be explicitly in k/k.

Comment (feiskyer) [reply contributors/devel/sig-release/cherry-picks.md:67]: yea, I mean cloud provider specific annotations here.

Comment (feiskyer) [reply contributors/devel/sig-release/cherry-picks.md:67]: let me add this into list above

Comment (feiskyer) [reply contributors/devel/sig-release/cherry-picks.md:67]: Updated per above discussion. @tpepper could you have a look again?

File: contributors/devel/sig-release/cherry-picks.md [hunk 2]
```diff
@@ -61,7 +61,13 @@ If the change is in cloud provider-specific platform code (which is in the
...
+- Serious bugs or broken scenarios (because of cloud platform changes) are fixed, e.g.
+  - new machine types are added
+  - new properties are required from Cloud API, which break existing features
```
[the remark this answers is not in the document]
Comment (feiskyer) [reply contributors/devel/sig-release/cherry-picks.md:70]: @tpepper  this issue would happen when users enable a new feature via latest cloud provider API version. While k/k vendor an older cloud provider API, the new feature is not supported there. Hence, k/k's cloud provider API invoking may be failed.

**Comment by nikhita** [comment]
:wave: friendly ping. Is this something we'd still like to get in?

**Comment by feiskyer** [comment]
close since is not needed anymore

**Pull request closed without merging**